internal/codeintel/lockfiles/go.go

```diff
@@ -0,0 +1,48 @@
+package lockfiles
+
+import (
+	"bufio"
+	"io"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf/reposource"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+//
+// go.sum
+//
+
+func parseGoSumFile(r io.Reader) ([]reposource.PackageDependency, error) {
+	var (
+		errs errors.MultiError
+
+		// In some cases, two checksums occur for both the package itself and the go.mod
+		// file of the same version.
+		deps = make(map[string]reposource.PackageDependency)
+	)
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		s := strings.Fields(line)
+		if len(s) < 3 {
```

Comment:
```suggestion
		if len(strings.Fields(line)) < 3 {
```

Nit: Temporarily binding `s` makes it look like `s` may be used again later.

Also, are there well-formed `go.sum` files which have fewer than 3 fields here, or is this just to be on the safer side?

Reply:
Good catch!

> Also, are there well-formed go.sum files which have fewer than 3 fields here, or is this just to be on the safer side?

A line in go.sum in current spec will always have 3 fields AFIAK, which are name, version and the checksum.